internal/xray/application.go

```diff
@@ -43,7 +45,16 @@ func (a *Application) Render(ctx context.Context, ns string, o interface{}) erro
 		gvr := gvkToGvr(res.GroupVersionKind())
 		switch gvr.String() {
 		case "apps/v1/deployments":
```

Comment:
I totally agree with using the registry for dynamic expansion of the tree & I actually tried it.

Unfortunately, the registry depends on `xray`, so I can't depend on it, otherwise I create a dependency cycle... What would you suggest to avoid this? Moving the `xray` types to a sub-package maybe?

Also, `applications` can manage themselves recursively ("app of apps" approach), so we'll need to keep track of rendered objects in the tree in order to avoid infinite loops.